src/readers/morphologyASC.cpp

```diff
@@ -151,33 +150,43 @@ class NeurolucidaParser
     /*
       Add the last point of parent section to the beginning of this section
       if not already present.
+      See https://github.com/BlueBrain/MorphIO/pull/221
+
+      The diameter is taken from the child section next point as does NEURON.
+      Here is the spec:
+      https://bbpteam.epfl.ch/project/issues/browse/NSETM-1178?focusedCommentId=135030&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-135030
+
+      In term of diameters, the result should look like the right picture of:
+      https://github.com/BlueBrain/NeuroM/issues/654#issuecomment-332864540
 
      The idea is that these two structures should represent the same morphology:
 
-     (3 -8 0 2)     and          (3 -8 0 2)
-     (3 -10 0 2)                 (3 -10 0 2)
+     (3 -8 0 5)     and          (3 -8 0 5)
+     (3 -10 0 5)                 (3 -10 0 5)
      (                           (
-       (0 -10 0 2)                 (3 -10 0 2)  <-- duplicate parent point
-       (-3 -10 0 2)                (0 -10 0 2)
+       (0 -10 0 2)                 (3 -10 0 2)  <-- duplicate parent point, note that the
+       (-3 -10 0 2)                (0 -10 0 2)      diameter is 2 and not 5
        |                           (-3 -10 0 2)
        (6 -10 0 2)                 |
-       (9 -10 0 2)                 (3 -10 0 2)  <-- duplicate parent point
-     )                             (6 -10 0 2)
+       (9 -10 0 2)                 (3 -10 0 2)  <-- duplicate parent point, note that the
+     )                             (6 -10 0 2)      diameter is 2 and not 5
                                    (9 -10 0 2)
                                  )
      */
-    void insertLastPointParentSection(int32_t parentId, morphio::Property::PointLevel& properties) {
+    void insertLastPointParentSection(int32_t parentId,
+                                      morphio::Property::PointLevel& properties,
+                                      std::vector<morphio::floatType>& diameters) {
         if (parentId < 0)  // Discard root sections
             return;
         auto parent = nb_.section(static_cast<unsigned int>(parentId));
         auto lastParentPoint = parent->points()[parent->points().size() - 1];
-        auto lastParentDiameter = parent->diameters()[parent->diameters().size() - 1];
+        auto childSectionNextDiameter = diameters[0];
```

Comment:
Out of curiousity, where is diameters checked to have size >= 1?

Reply:
`insertLastPointParentSection` is called by `_create_soma_or_section` which is called by `parse_neurite_section` with the following code which discards empty sections:

```cpp
if (!points.empty()) {
    _create_soma_or_section(token, parent_id, points, diameters);
}
```

Reply:
cool, thanks; and diameters & points is always mirrored?

Reply:
Yes. And I'm confident that this would be caught by the unit tests otherwise. https://github.com/BlueBrain/MorphIO/blob/master/tests/test_2_neurolucida.py is full of edge cases.